### PR TITLE
[SPARK-30627][SQL] Disable all the V2 file sources by default

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Disable all the V2 file sources in Spark 3.0 by default.

### Why are the changes needed?

There are still some missing parts in the file source V2 framework:
1. It doesn't support reporting file scan metrics such as "numOutputRows"/"numFiles"/"fileSize" like `FileSourceScanExec`. This requires another patch in the data source V2 framework. Tracked by [SPARK-30362](https://issues.apache.org/jira/browse/SPARK-30362)
2. It doesn't support partition pruning with subqueries(including dynamic partition pruning) for now. Tracked by [SPARK-30628](https://issues.apache.org/jira/browse/SPARK-30628)

As we are going to code freeze on Jan 31st, this PR proposes to disable all the V2 file sources in Spark 3.0 by default.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @cloud-fan @rdblue  @gatorsmile

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -1728,7 +1728,7 @@ object SQLConf {
       "implementation class names for which Data Source V2 code path is disabled. These data " +
       "sources will fallback to Data Source V1 code path.")
     .stringConf
-    .createWithDefault("kafka")
+    .createWithDefault("kafka,parquet,orc,json,csv,text,avro")
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Shall we set this alphabetically at this time, @gengliangwang ?

Comment (gengliangwang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: ok, sure

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. (Pending Jenkins. Hopefully, we didn't miss anything in test suites)

**Review by gatorsmile — APPROVED** [review]
LGTM pending Jenkins

File: sql/core/src/test/scala/org/apache/spark/sql/connector/FileDataSourceV2FallBackSuite.scala [hunk 2]
```diff
@@ -86,6 +87,11 @@ class FileDataSourceV2FallBackSuite extends QueryTest with SharedSparkSession {
   private val dummyReadOnlyFileSourceV2 = classOf[DummyReadOnlyFileDataSourceV2].getName
   private val dummyWriteOnlyFileSourceV2 = classOf[DummyWriteOnlyFileDataSourceV2].getName
 
+    override protected def sparkConf: SparkConf =
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/FileDataSourceV2FallBackSuite.scala]: indentation?

Comment (gengliangwang) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/FileDataSourceV2FallBackSuite.scala]: Updated, thanks!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM again. Thank you, @gengliangwang and @gatorsmile .
All tests passed and the last commit is only indentation changes.
Merged to master.

**Pull request closed without merging**